### DISCUSSION TITLE
How to configure "LSP server configuration schema"?

### DISCUSSION THREAD

**Discussion by fritzrehde** [mason-org/mason.nvim#638] [Q&A]
Hi, I'm very new to neovim LSP and mason. I am using nvim-cmp, mason.nvim, mason-lspconfig.nvim and nvim-lspconfig.
When calling `:Mason` in neovim, I see all available and installed LSPs, and when clicking on one I see some config options. I want to configure some of the options in "LSP server configuration schema", but have absolutely no clue how to do that? Do I do this with Mason, nvim lsp or the LS itself (how though?)?
Thanks!

**Comment by williamboman** [comment]
Hey! So what's presented in the `:Mason` window is a read-only overview of what settings the server has defined in its schema. How you modify these settings depend on how you set up your servers. With `lspconfig` it's done via the `settings` table during setup (`:h lspconfig-setup`), for example like so:

```lua
require("lspconfig").pyright.setup {
    settings = {
        pyright = {
            disableLanguageServices = true,
            disableOrganizeImports  = true,
        }
    }
}
```

**Reply by kaffarell** [reply]
Would it be possible to have this integrated in `:Mason`? Would be really cool if I could press `e` for example and edit the values. My use case is that I want to edit the feature-flags used by rust-analyzer quickly.
